examples/generatorOptions.md

```diff
@@ -0,0 +1,127 @@
+# Generator Options
+
+Kustomize provides options to modify the behavior of ConfigMap and Secret generators. These options include
+ 
+ - disabling hash appended to the generated resources
+ - adding labels to generated resources
+ - adding annotations to generated resources
+ - changing shell and arguments for getting data from commands
+ - changing timeout for executing commands
+ 
+This demo shows how to use these options. First create a workspace.
+```
+DEMO_HOME=$(mkdir -d)
+```
+
+Create a kusotmization and add a ConfigMap generator to it.
+
+<!-- @createCMGenerator @test -->
+```
+cat > $DEMO_HOME/kustomization.yaml << EOF
+configMapGenerator:
+- name: my-configmap
+  literals:	
+  - foo=bar
+  - baz=qux
+EOF
+```
+
+## disabling hash
+By default, ConfigMap generator adds a suffix to the names of generated ConfigMaps.
+The suffix is a hash of the ConfigMap content. Add following generatorOptions to kustomization can disable appending the suffix.
+
+<!-- @disableHash @test -->
+```
+cat >> $DEMO_HOME/kustomization.yaml << EOF
+generatorOptions:
+ disableHash: true
+EOF
+```                              
+Run `kustomize build`, the generated ConfigMap name will not have a hash suffix.
+
+> ```
+> apiVersion: v1
+> data:
+>   baz: qux
+>   foo: bar
+> kind: ConfigMap
+> metadata:
+>   name: my-configmap
+> ```
+
+## adding labels
+By specifying labels in generatorOptions, all generated resources will have those labels.
+<!-- @addlabel @test -->
+```
+sed -i '/generatorOptions/,$d' $DEMO_HOME/kustomization.yaml
+
+cat >> $DEMO_HOME/kustomization.yaml << EOF
+generatorOptions:
+  labels:
+    kustomize.generated.resources: somevalue
+EOF
+```
+Run `kustomize build`, the generated ConfigMap will have the label `kustomize.generated.resources`.
+
+> ```
+> apiVersion: v1
+> data:
+>   baz: qux
+>   foo: bar
+> kind: ConfigMap
+> metadata:
+>   labels:
+>     kustomize.generated.resources: somevalue
+>   name: my-configmap-g698dg7tfc
+> ```
+
+## adding annotations
```

Comment:
The rest of this file feels like it could be dropped or compressed to one example.  Its simple and repetitive.

Reply:
How about compressing all of them into one example: disabling name suffix, adding label or annotation, changinger timeout and shell?

Reply:
sgtm